code/Test_definitions/Geofencing.feature

```diff
@@ -0,0 +1,179 @@
+
+  @geofencing_subscriptions_13_no_authorization_header
+  Scenario: No Authorization header
+    Given a valid subscription request body and header "Authorization" is not available
+    When the  request "createSubscription" is sent 
+    Then the response status code is 401
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+
+  @geofencing_subscriptions_14_expired_access_token
+  Scenario: Expired access token
+    Given a valid subscription request body and header "Authorization" is expired
+    When the  request "createSubscription" is sent
+    Then the response status code is 401
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+
+  @geofencing_subscriptions_15_invalid_access_token
+  Scenario: Invalid access token
+    Given a valid subscription request body and header "Authorization" set to an invalid access token
+    When the  request "createSubscription" is sent
+    Then the response status code is 401
+    And the response header "Content-Type" is "application/json"
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
+    And the response property "$.message" contains a user friendly text
+
+   @geofencing_subscriptions_16_sink_credentials
+  Scenario:   Validate that the subscribed events are received in the sink, with the right sinkCredential, for those situations specified in the API.
+    Given a valid subscription request body with "sink" and "sinkcredentials"  
+    When the  request "createSubscription" is sent  
+    Then the response  code is 201 and subscription is created
+    Then the  event details are observed on notifications-url
+    And the subscribed event  received on notifications-callback-url & sink credentials are as expected
+
+ @geofencing_subscriptions_17_subscriptionExpireTime
+  Scenario:   For subscriptions that provide subscriptionExpireTime, validate that the subscribed events are not longer received after the expiration time.
+    Given a valid subscription request body and for less $.duration
+    When the request "createSubscription" is sent
+    Then the subscription is expired
+    And no event notifications received on callback-url when device enters or leaves geofence
+    
+  @geofencing_subscriptions_18_subscriptionMaxEvents
+   Scenario: For subscriptions that provide subscriptionMaxEvents, validate that the subscribed events are not longer received after the maximum events limit is reached.
+    Given a valid subscription request body 
+    When the  request "createSubscription" is sent
+    Then the  maxevents are already triggered for available subscription
+    And no event notifications received on callback-url when device enters or leaves geofence
```

Comment:
or me we should replace your line 171 with:

```
 And The callback notification application receives subscription-ends event at provided callbackUrl
 And notification body complies with the OAS schema at "##/components/schemas/CloudEvent"
 And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends
```"

indeed in this case a subscription-ends has to be sent to the callback url.